packages/shared/src/hooks/streaks/useReadingStreak.ts

```diff
@@ -39,48 +41,35 @@ interface UserReadingStreak {
 export const useReadingStreak = (): UserReadingStreak => {
   const { user, isLoggedIn } = useAuthContext();
   const { optOutReadingStreak, loadedSettings } = useContext(SettingsContext);
+  const { checkHasCompleted } = useActions();
   const queryClient = useQueryClient();
   const queryKey = generateQueryKey(RequestKey.UserStreak, user);
 
-  const { data: streak, isPending } = useQuery({
+  const { data: streak, isPending: isPendingQuery } = useQuery({
     queryKey,
     queryFn: getReadingStreak,
     staleTime: StaleTime.Default,
     enabled: isLoggedIn,
   });
 
-  const { mutate: updateStreakConfig } = useMutation<
-    UserStreak,
-    ResponseError,
-    UpdateReadingStreakConfig
-  >({
-    mutationFn: (params) =>
-      gqlClient.request(UPDATE_STREAK_COUNT_MUTATION, params),
-
-    onMutate: ({ weekStart }) => {
-      queryClient.cancelQueries({ queryKey });
+  const { mutate: updateStreakConfig, isPending: isPendingMutation } =
+    useMutation<UserStreak, ResponseError, UpdateReadingStreakConfig>({
+      mutationKey: generateQueryKey(RequestKey.UserStreak, user, 'config'),
+      mutationFn: async (params) =>
+        await gqlClient.request(UPDATE_STREAK_COUNT_MUTATION, params),
```

Comment:
I think this returns user streak after update so we can use it to update `queryKey` without refetching again in onSuccess?